### ISSUE TITLE
5 plugins extend async class and hook to the same "save_post" hook what will happen in this situation 

### ISSUE THREAD

**Issue by ahmedofali** [techcrunch/wp-async-task#17]
I Need to know if it will send 5 remote post requests or not ?

**Comment by ericmann** [comment]
Yes, it will send 5 remote post requests.

Every instance of `WP_Async_Task` hooks its own `launch` routine to the specified hook, in this case, "save_post." When "save_post" fires, WordPress will invoke each `launch` routine individually, which will each then hook the instance's `launch_on_shutdown` function to WordPress' "shutdown" action.

When WordPress completes the request, it triggers "shutdown," and each `WP_Async_Task` instance will trigger its own remote POST back to the server.

## Why the duplication?

In your question, you point out that 5 separate plugins are all doing the same thing and hooking in the same way. Conceptually, that makes it feel like they should group the work together and fire a _single_ async event. However, that wouldn't be a good idea.

In this event, it's merely a coincidence that all of the plugins are hooking to the same event. The reality is that each plugin is specifying an event that should take place asynchronously (or "in the background without blocking anything else WordPress is doing"). They have no knowledge of one another, so they couldn't coordinate in the first place.

Also, async tasks are intended for long-running jobs or any data manipulation that would otherwise lead to long timeouts or hangs in normal operation. Said another way, if the functions being invoked run quickly, they should just use a _normal_ "save_post" action, not an asynchronous one. If there are multiple actions happening asynchronously (i.e. the 5 separate plugins), then through an async hook they'll all run _in parallel_ and complete much faster.

If they were normal "save_post" events, or if the async tasks were somehow grouped together, this kind of concurrent operation would not be possible.

## In Summary

> I Need to know if it will send 5 remote post requests or not ?

Yes, it will send 5 remote post requests. This is by design.
